### PR TITLE
dev: adding ESLint and Prettier (for fire-and-forget code-consistency)

### PR DESCRIPTION
**Pull request by nemoDreamer**
Y U NO LINT, BRO?!

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		curly: ['error', 'all'],
+		'prettier/prettier': [
+			'error',
+			{
+				printWidth: 120,
+				semi: false,
+				singleQuote: true,
+				trailingComma: 'none', // ideally 'all'...
+				tabWidth: 1,
+				useTabs: true
```
Comment (nemoDreamer) [inline .eslintrc.js:27]: I noticed you mainly used tabs, but if you want to use 2 spaces (hint hint), set:

```js
{
  tabWidth: 2,
  useTabs: false
}
```

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	},
+	rules: {
+		curly: ['error', 'all'],
+		'prettier/prettier': [
+			'error',
+			{
+				printWidth: 120,
+				semi: false,
+				singleQuote: true,
+				trailingComma: 'none', // ideally 'all'...
```
Comment (nemoDreamer) [inline .eslintrc.js:25]: `trailingComma: 'all'` results in a large initial diff, but smaller diffs later on, as adding things to hashes/arrays doesn't require you to add a comma to the end of the previous line.

**Comment by nemoDreamer** [comment]
If you use Atom, install `linter` and `linter-eslint`, and turn on "Lint on Save", then you'll never have to worry about indentation/flow/formatting, and it'll tell you when you're importing modules or defining stuff that you aren't using, etc...

**Comment by forrestSoft** [comment]
yey. lint

**Pull request merged**